docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,135 @@
+# APIMan Integration
+
+## Abstract
+This proposes a design for integrating [APIMan](http://www.apiman.io/) with OpenShift in order to provide  micro-services governance for API service providers.
+
+## Motivation
+Providers of services may have the need to control who and how their services are consumed.  This will be realized as policies defined by providers to control their service in any number of ways (e.g security, throttling/quota, billing and metrics)<sup>[1](#r1)</sup>. 
+
+## Constraints and Assumptions
+### Authentication
+The OAuth token identifying a user in OpenShift will be utilized by APIMan to associate service policy to their identity.  This will ensure a user's identity is ubiquitous across the cluster when interacting with OpenShift and the APIMan management console.
+
+### Deployment
+#### Communication
+APIMan components (e.g. gateway, management interface) integrated with the cluster will utilize mutual TLS for internal communication.
+
+#### Storage
+APIMan is capable of using several backends (e.g. ElasticSearch, Postgresql) to store configuration, policy, and metrics.  The initial integration will utilize an ElasticSearch cluster dedicated to APIMan.  The ElasticSearch image will be shared with aggregated logging.  The [ACL plugin](https://github.com/fabric8io/openshift-elasticsearch-plugin)<sup>[2](#r2)</sup> that is deployed as part of aggregated logging will be disabled in the APIMan ElasticSearch deployment. Additionally, the ElasicSearch index management<sup>[3](#r3)</sup>  will be modified so that it will not cull APIMan data.
+
+
+Cluster administrators can configure APIMan to use an alternate, off cluster ElasticSearch instance if desired.  Some organizations may have an existing instance of ElasticSearch cluster that already defines policy, and administrators may desire to use it instead of the one provided with aggregated logging.
```

Comment:
In the context of apiman, ES_COPY is not relevant since its a fluentd plugin.  If our deployer is not capable of configuring apiman to write to a non-secure ES instance, then we need to make it capable.